packages/flutter_tools/lib/src/commands/trace.dart

```diff
@@ -55,10 +55,14 @@ class TraceCommand extends FlutterCommand {
   Future<Null> runCommand() async {
     int observatoryPort = int.parse(argResults['debug-port']);
 
+    // TODO(danrubel): this will break if we move to the new observatory URL
+    // See https://github.com/flutter/flutter/issues/6843
+    Uri observatoryUri = Uri.parse('http://127.0.0.1:$observatoryPort');
```

Comment:
please file an issue that the trace command doesn't scrape the Observatory uri from device logs and add a TODO(issue#) here.